### PR TITLE
Updated te and te-types with RFC8345 identifiers

### LINKED ISSUE #194: A question about the design of TE TOPOLOGY objects' identifier
Different identifiers for networks, nodes, links and termination points in network topology and TE topology models

For example:
- node-id is an URI
- te-node-id is a dotted-quad

Proposal: update the TE types to be an union with URI; for example: te-node-id is an union between dotted-quad and URI

See: https://mailarchive.ietf.org/arch/msg/teas/SdQbL_yOavuqQxSJqOvJK_d7qTg/

### PR DESCRIPTION
**Pull request by italobusi**
Proposed changes to fix #194 

Co-Authored-By: aguoietf <83043060+aguoietf@users.noreply.github.com>

### CODE REVIEW

File: ietf-te.yang [hunk 1]
```diff
@@ -606,29 +621,46 @@ module ietf-te {
...
-        "TE tunnel destination node identifier.";
+        "The TE tunnel end-point TE node identifier";
     }
-    leaf src-tunnel-tp-id {
+    leaf tunnel-tp-id {
+      when "../node-id or ../te-node-id" {
+        description
+          "The TE tunnel termination point identifier is local to 
+          a node";
+      }
```
[the remark this answers is not in the document]
Comment (tsaad-dev) [reply ietf-te.yang:643]: Node needs to be identified and then the TTP-ID (binary) can be set

File: ietf-te.yang [hunk 2]
```diff
@@ -681,21 +713,42 @@ module ietf-te {
...
+          type nt:tp-id;
+          description
+            "The remote link termination point identifier.";
+        }
         leaf remote-te-node-id {
           type te-types:te-node-id;
           description
             "Remote TE node identifier.";
         }
+        leaf network-id {
```
Comment (tsaad-dev) [inline ietf-te.yang:746]: We understand the grouping te-toplogy-identifier in both TE tunnel and TE topology.
Can we modify the:
uses te-types:te-topology-identifier and augment an additional leaf network here in TE tunnel?

Comment (tsaad-dev) [reply ietf-te.yang:746]: The network-id identifies the the network topology.
The te-topology-identifier container identifies the TE network topology. 
Both can exist independently.

File: ietf-te.yang [hunk 3]
```diff
@@ -681,21 +713,42 @@ module ietf-te {
...
+          type nw:network-id;
+          description
+            "The network topology identifier where the hierarchical 
+            link supported by this TE tunnel is instantiated.";
+        }
         uses te-types:te-topology-identifier {
           description
             "The topology identifier where the hierarchical link
```
Comment (tsaad-dev) [inline ietf-te.yang:754]: Please change the description to 'The TE topology identifier...

Comment (italobusi) [reply ietf-te.yang]: Fixed https://github.com/tsaad-dev/te/pull/229/commits/4293a72724a6a311f05f7a5225cadac71dc4b542

**Pull request merged**